From: John Brandwood <john.brandwood@telzey.com>
Date: Fri, 9 May 2025 11:24:24 -0400
Subject: Stop using "true" and "false" as label names in system.c, it breaks
 modern C!
Origin: https://github.com/jbrandwood/mcpp.git/commit/1d65736ca2c10328916ed6c98e6422a784da187a

---
 src/system.c | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/src/system.c b/src/system.c
index cb21e99..a6ff95b 100644
--- a/src/system.c
+++ b/src/system.c
@@ -3450,7 +3450,7 @@ search:
     if (! fullname)                 /* Non-existent or directory    */
         return  FALSE;
     if (standard && included( fullname))        /* Once included    */
-        goto  true;
+        return  TRUE;
         
     if ((max_open != 0 && max_open <= include_nest)
                             /* Exceed the known limit of open files */
@@ -3477,12 +3477,12 @@ search:
         if ((fp = fopen( fullname, "r")) == NULL) {
             file->fp = fopen( cur_fullname, "r");
             fseek( file->fp, file->pos, SEEK_SET);
-            goto  false;
+            goto  failed;
         }
         if (max_open == 0)      /* Remember the limit of the system */
             max_open = include_nest;
     } else if (fp == NULL)                  /* No read permission   */ 
-        goto  false;
+        goto  failed;
     /* Truncate buffer of the includer to save memory   */
     len = (int) (file->bptr - file->buffer);
     if (len) {
@@ -3529,9 +3529,8 @@ search:
     if (mkdep && ((mkdep & MD_SYSHEADER) || ! infile->sys_header))
         put_depend( fullname);          /* Output dependency line   */
 
-true:
     return  TRUE;
-false:
+failed:
     free( fullname);
     return  FALSE;
 }
-- 
2.51.0

